src/main/java/hudson/plugins/s3/S3BucketPublisher.java

```diff
@@ -119,7 +127,9 @@ public static S3Profile getProfile(String profileName) {
     }
 
     private void log(final PrintStream logger, final String message) {
-        logger.println(StringUtils.defaultString(getDescriptor().getDisplayName()) + ' ' + message);
+        if(! suppressLogging) {
```

Comment:
Sorry, but I don't think that this is a right place for suppressing. Probably, you want to suppress only messages that's related to pushing files to s3 - not all messages (including warning and errors).
So, I can see two solutions:
- Move this check to line 185/195
- Introduce three different levels: errors (after such `log` you can see `return` statement or `setResult(failed)`), warning ('no files found', 'skip publishing', etc.) and info (this one you want to skip).